### PR TITLE
services/horizon: Add --max-assets-per-path-request flag

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [X] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [X] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Introduce a flag that sets the max number of assets to consider during strict-send or strict-receive.

### Why

The maximum (15) is hardcoded, and making it configurable grants users some flexibility.

### Known limitations

N/A

### CODE REVIEW

File: services/horizon/CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,8 @@ file. This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Changes
 
-* Add an endpoint that allows querying for which liquidity pools an account is participating in 
+* Add a new feature flag `--max-assets-per-path_request` (`15` by default) that sets the number of assets to consider for strict-send and strict-recieve ([4046](https://github.com/stellar/go/pull/4046))
```
Comment (tamirms) [inline services/horizon/CHANGELOG.md:10]: nit: I would replace "feature flag" with "horizon flag"

**Review by tamirms — APPROVED** [review]

File: services/horizon/CHANGELOG.md (hunk 1 above)
Comment (bartekn) [reply services/horizon/CHANGELOG.md:10]:
```suggestion
* Add a new feature flag `--max-assets-per-path-request` (`15` by default) that sets the number of assets to consider for strict-send and strict-recieve ([4046](https://github.com/stellar/go/pull/4046))
```

**Review by bartekn — APPROVED** [review]

**Comment by marcinx** [comment]
Is it possible to bump up the default though? 15 seems arbitrary and looking at response times of requests with 15 assets, the pathfinder should be able to handle a multiple of that easily.

Trying to avoid situations where we use, e.g. 100, on our deployment and are not able to fall back to SDF Horizon because of a much lower constraint.

File: services/horizon/CHANGELOG.md (hunk 1 above)
Comment (erika-sdf) [reply services/horizon/CHANGELOG.md]: Done

**Comment by ire-and-curses** [comment]
@marcinx We're not going to change this number for SDF's Horizon in the near future. It's unclear what the impact of raising the number is for our load. We have a large number of requests, and even small performance changes have large consequences when multiplied across all our traffic. We would need to simulate that before making any tweaks here.

As far as fall back goes, if your use case requires this very large path, it would be a great idea to consider a backup or secondary cluster for the scenarios you mentioned. It is dangerous in general, and not recommended, to rely on the SDF Horizon instance for production use.

**Comment by marcinx** [comment]
@ire-and-curses Okay, thanks for the reply. That's a good idea and I had a setup with a separate Horizon instance just for path payments in mind already. Or maybe we'll come up with a way to iterate over path requests in our code eventually.

**Pull request merged**